### PR TITLE
[LRFID] Add support for Nexkey/Nexwatch

### LINKED ISSUE #2675: Honeywell NexKey RFID cards
### Describe the enhancement you're suggesting.

I'd like to see support for Honeywell NexKey RFID cards.  My company has a system that we can currently neither replace nor modify, and we're looking to be able to copy cards with access.  I'm attaching raw reads from three different cards.

[NexKeyRAW.zip](https://github.com/flipperdevices/flipperzero-firmware/files/11488756/NexKeyRAW.zip)

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mauimauer**
Add support for  Nexkey/Nexwatch. Fixes #2675 

# What's new

New protocol for lrfid
- Nexkey/Nexwatch (PSK)

# Verification 

- Use the LF-RFID application as usual, try to read, emulate, write new protocol.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Review by DrZlo13 — DISMISSED** [review]

**Review by skotopes — APPROVED** [review]

File: lib/lfrfid/protocols/protocol_nexwatch.c [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+}
+
+static void protocol_nexwatch_descramble(uint32_t* id, uint32_t* scrambled) {
+    // 255 = Not used/Unknown other values are the bit offset in the ID/FC values
+    const uint8_t hex_2_id[] = {31, 27, 23, 19, 15, 11, 7, 3, 30, 26, 22, 18, 14, 10, 6, 2,
+                                29, 25, 21, 17, 13, 9,  5, 1, 28, 24, 20, 16, 12, 8,  4, 0};
+
+    *id = 0;
+    for(uint8_t idx = 0; idx < 32; idx++) {
+        if(hex_2_id[idx] == 255) continue;
```
Comment (skotopes) [inline lib/lfrfid/protocols/protocol_nexwatch.c]: this expression is always false

**Review by skotopes — CHANGES_REQUESTED** [review]

File: lib/lfrfid/protocols/protocol_nexwatch.c (hunk 1 above)
Comment (xMasterX) [reply lib/lfrfid/protocols/protocol_nexwatch.c]: What I found:

Screenshot 2023-05-26 at 20 00 14

-----

Issue comes from proxmark code

It has same check thats always false 

I think if it works with current `hex_2_id` table, this `if(hex_2_id[idx] == 255) continue;` can be safely removed without any issues

Comment (skotopes) [reply lib/lfrfid/protocols/protocol_nexwatch.c]: yes please, remove it and open issue in PM repository

Comment (mauimauer) [reply lib/lfrfid/protocols/protocol_nexwatch.c]: Done

**Review by skotopes — APPROVED** [review]

**Pull request merged**